CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelIndices.h

```diff
@@ -0,0 +1,358 @@
+#ifndef CTPPS_PIXELINDICES_H
+#define CTPPS_PIXELINDICES_H
+
+#include <iostream>
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+/**
+ * Numbering of the pixels inside the readout chip (ROC).
+ * There is a column index and a row index.
+ * In the barrel the row index is in the global rfi direction (local X) and
+ * the column index is in the global z (local Y) direction.
+ * In the endcaps the row index is in the global r direction (local X) and
+ * the column index in the gloabl rfi (local Y) direction.
+ *
+ * In CTPPS  the row index is in the global y direction (local X) and
+ * the column index is in the global x (local Y) direction.
+ * The defaults are specific to 100*150 micron pixels.
+ * 
+ * Some methods are declared as static and can be used without class 
+ * instantiation. Others need the construstor to setup the Module size
+ * parameters. These parameters are only used for error checking.
+ * d.k. 10/2005
+ */
+
+/*
+
+  sensor side = bump bonding side = wire bonding side
+
+
+  ^   sim y
+  |
+  |
+  |
+  0-----------------------------------
+  |                |                  |
+  |        2      |       3         |
+  |                |                  0
+  --------------------------------------
+  0-----------------------------------
+  |                |                  |
+  |       1       |-------4---------|--------------> sim x        
+  |                |                  0
+  --------------------------------------
+  0-----------------------------------   ^
+  |                |                  |    | det cols 
+  |      0        |        5        |    |
+  |                |                  0   |
+  --------------------------------------   |
+  |
+  <------------------------------------------ 0,0
+  det rows x
+  0  beam
+
+*/
+
+namespace {
+ 
+  // The maximum number of ROCs in the X (row) direction per sensor.
+  const int maxROCsInX = 2;  //  
+  // The maximum number of ROCs in the Y (column) direction per sensor.
+  const int maxROCsInY = 3;  //
+  // The nominal number of double columns per ROC is 26. 
+  const int DColsPerROC = 26; 
+  // Default ROC size 
+  const int ROCSizeInX = 80;  // ROC row size in pixels 
+  const int ROCSizeInY = 52;  // ROC col size in pixels 
+  // Default DET barrel size 
+  const int defaultDetSizeInX = 160;  // Det row size in pixels (2 ROCs) 
+  const int defaultDetSizeInY = 156; //  Det col size in pixels (3 ROCs) 
+
+  // Check the limits
+  const bool CTPPS_CHECK_LIMITS = true;
```

Comment:
This looks like a debug flag and it would enable the compiler to optimize it's usage if it is declared `constexpr`.